### PR TITLE
Add build-flag guards to PR review checklist

### PR DESCRIPTION
**Pull request by petemill**

### CODE REVIEW

**Comment by petemill** [comment]
I guess both - the build flags need to be added for code related to those features because we already have code related to those features which are surrounded by `#if FLAG` guards. And the reason for introducing those in the first place, adn the related build config that switches those flags on and off, is I guess because someone decided that they should be disable-able. The extensions one is already because that is not enabled on android, but sure we should add an entry about platform-specific code.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -32,7 +32,13 @@
 
 - [ ] New files have MPL-2.0 license header.
 - [ ] Request a security/privacy review as needed.
-- [ ] Adequate test coverage exists to prevent regressions 
+- [ ] Adequate test coverage exists to prevent regressions
+- [ ] Build-flag guards are used for code that touches:
+    - Ads
```
Comment (bridiver) [inline .github/PULL_REQUEST_TEMPLATE.md]: also Wallet, Referrals, Greaselion and Translate. Maybe also a checkbox to update this list when new buildflags are added?

**Comment by bsclifton** [comment]
Closing as stale. I think we're wanting to get rid of some of the guards soon (ex: with rewards)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (bbondy) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: I'd suggest just don't try to be exhaustive here. This already misses a lot of them like crypto wallets, binance, gemini, ipfs, etc.

**Comment by bbondy** [comment]
> Closing as stale. I think we're wanting to get rid of some of the guards soon (ex: with rewards) 

BTW we're still keeping guards, but only when it needs to be gated by something that can't be compiled on a platform. 
Example: Crypto Wallets needs extensions, and extensions are not on Android. So that needs to be a build flag and not a about:flags

**Pull request closed without merging**